bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GenericItem.java

```diff
@@ -370,32 +369,7 @@ public void setCategory(@Nullable String category) {
 
     @Override
     public @Nullable StateDescription getStateDescription(@Nullable Locale locale) {
-        StateDescription result = null;
-        List<StateOption> stateOptions = Collections.emptyList();
-        if (stateDescriptionProviders != null) {
-            for (StateDescriptionProvider stateDescriptionProvider : stateDescriptionProviders) {
-                StateDescription stateDescription = stateDescriptionProvider.getStateDescription(this.name, locale);
-
-                // as long as no valid StateDescription is provided we reassign here:
-                if (result == null) {
-                    result = stateDescription;
-                }
-
-                // if the current StateDescription does provide options and we don't already have some, we pick them up
-                // here
-                if (stateDescription != null && !stateDescription.getOptions().isEmpty() && stateOptions.isEmpty()) {
-                    stateOptions = stateDescription.getOptions();
-                }
-            }
-        }
-
-        // we recreate the StateDescription if we found a valid one and state options are given:
-        if (result != null && !stateOptions.isEmpty()) {
-            result = new StateDescription(result.getMinimum(), result.getMaximum(), result.getStep(),
-                    result.getPattern(), result.isReadOnly(), stateOptions);
-        }
-
-        return result;
+        return stateDescriptionService.getStateDescription(this.name, locale);
```

Comment:
This could result in an NPE as stateDescriptionService is nullable (and a dynamic+optional dependency in ItemRegistryImpl).